src/Adapter/Order/OrderAmountUpdater.php

```diff
@@ -180,22 +180,25 @@ private function updateOrderTotals(Order $order, Cart $cart, int $computingPreci
         $order->total_discounts_tax_excl = (float) abs($cart->getOrderTotal(false, Cart::ONLY_DISCOUNTS, $orderProducts, $carrierId, false, $this->keepOrderPrices));
         $order->total_discounts_tax_incl = (float) abs($cart->getOrderTotal(true, Cart::ONLY_DISCOUNTS, $orderProducts, $carrierId, false, $this->keepOrderPrices));
 
+        // We set $useEcotax to false because
+        //    $cart->getOrderTotal will fetch from order details which have already ecotax in their prices
+        $useEcotax = false;
         // We should always use Cart::BOTH for the order total since it contains all products, shipping fees and cart rules
         $order->total_paid = Tools::ps_round(
-            (float) $cart->getOrderTotal(true, Cart::BOTH, $orderProducts, $carrierId, false, $this->keepOrderPrices),
+            (float) $cart->getOrderTotal(true, Cart::BOTH, $orderProducts, $carrierId, false, $this->keepOrderPrices, $useEcotax),
```

Comment:
About the bug that @khouloudbelguith found, I guess you need to perform the same modifications in `updateOrderInvoices` And you'll probably need to add behat tests to check the invoice values as well

Reply:
That works. And behat tests added ofc.